### PR TITLE
CUDA 12+ support with Nvbit

### PR DESCRIPTION
**Pull request by ejchung0406**
- Added NVBit Tracer to `tools/`
- NVBit trace reader to `src/`

### CODE REVIEW

File: bin/params.in [hunk 1]
```diff
@@ -1,85 +1,112 @@
...
 # DRAM
-dram_bus_width 4
+dram_num_mc 6
+dram_bus_width 8
 dram_column 11
 dram_activate 25
 dram_precharge 10
 dram_num_banks 16
-dram_num_channel 8
+dram_num_channel 1
```
Comment (jaewon-lee-github) [inline bin/params.in:84]: Why dram_num_channel is 1?

File: bin/trace_file_list [hunk 2]
```diff
@@ -1,2 +1,4 @@
 1
-../sst-unit-test/traces/cachesize_1/trace.txt
\ No newline at end of file
+/fast_data/trace/nvbit/vectoradd/kernel/Trace.txt
```
Comment (jaewon-lee-github) [inline bin/trace_file_list]: Macsim should work other than our server. Can you set the default trace_file_list to pre-existing traces? We can just put simple vadd nvbit trace in bin folder and point it out.

File: bin/params.in (hunk 1 above)
Comment (jaewon-lee-github) [inline bin/params.in]: Why dram_num_channel is changed from 8 to 1? Doesn't it matter in default setting?

File: src/global_defs.h [hunk 3]
```diff
@@ -159,7 +159,9 @@ void delete_store_hash_entry_wrapper(map_c *map, uop_c *uop);
 ///////////////////////////////////////////////////////////////////////////////////////////////
 // Global definitions
 
-#define MAX_TR_OPCODE_NAME GPU_OPCODE_LAST
+// JAEWON FIXME GPU
+//#define MAX_TR_OPCODE_NAME GPU_OPCODE_LAST
```
Comment (jaewon-lee-github) [inline src/global_defs.h]: Can you delete this code? ;)

File: src/process_manager.cc [hunk 4]
```diff
@@ -263,6 +253,8 @@ void process_manager_c::create_thread_node(process_s *process, int tid,
 
   // block id assignment in case of multiple applications
   int block_id = start_info->m_thread_id >> BLOCK_ID_SHIFT;
+  // inside parenthesis is the unique block id 
+  // multi_key_map_c::find(key1, key2)
```
Comment (jaewon-lee-github) [inline src/process_manager.cc]: can you delete this if it is not needed?

File: src/process_manager.h [hunk 5]
```diff
@@ -181,6 +181,7 @@ typedef struct thread_s {
   int m_orig_block_id; /**< original block id from a trace*/
   int m_orig_thread_id; /**< adjusted block id */
   int m_block_id; /**< block id */
+  uint64_t m_trace_id;
```
Comment (jaewon-lee-github) [inline src/process_manager.h]: Is this trace_id or thread_id?

**Review by jaewon-lee-github — CHANGES_REQUESTED** [review]

File: bin/trace_file_list (hunk 2 above)
Comment (ejchung0406) [reply bin/trace_file_list]: Added vectormultadd example in `sst-unit-test/traces/nvbit`

File: bin/params.in (hunk 1 above)
Comment (ejchung0406) [reply bin/params.in]: Reverted to 8

File: src/global_defs.h (hunk 3 above)
Comment (ejchung0406) [reply src/global_defs.h]: sure

File: src/process_manager.cc (hunk 4 above)
Comment (ejchung0406) [reply src/process_manager.cc]: i think it is for explaining using the multi-key map

File: src/process_manager.h (hunk 5 above)
Comment (ejchung0406) [reply src/process_manager.h]: I deleted `m_trace_id`, it was never used

**Review by jaewon-lee-github — APPROVED** [review]
Nice work!

**Pull request merged**